src/quo2/components/messages/system_message.cljs

```diff
@@ -0,0 +1,145 @@
+  [rn/view {:margin-left 6}
+   [text/text {:size  :label
+               :style {:color          (get-color :time)
+                       :text-transform :none}}
+    timestamp-str]])
+
+(defn sm-icon [icon]
+  [rn/view {:align-items  :center
+            :margin-right 8}
+   [icon-avatar/icon-avatar {:size  :large
+                             :icon  icon
+                             :color :primary
+                             :opacity 5}]])
+
+(defmulti sm-render :type)
+
+(defmethod sm-render :deleted [_]
+  [rn/view {:align-items     :center
+            :justify-content :space-between
+            :flex            1
+            :flex-direction  :row}
+   [rn/view {:align-items    :center
+             :flex-direction :row}
+    [sm-icon :main-icons/placeholder20]
+    [text/text {:size  :paragraph-2
+                :style {:color        (get-color :text)
+                        :margin-right 5}}
+     (i18n/label :message-deleted-for-everyone)]]
+   [button/button {:size   24
+                   :before :main-icons/timeout
+                   :type   :grey} (i18n/label :undo)]])
+
+(defmethod sm-render :added [{:keys [mentions timestamp-str]}]
+  [rn/view {:align-items    :center
+            :flex-direction :row}
+   [sm-icon :main-icons/placeholder20]
+   [rn/view {:margin-right 4}
+    [user-avatar/user-avatar {:status-indicator? false
+                              :online?           false
+                              :size              :xxxs
+                              :profile-picture   (:image (first mentions))
+                              :ring?             false}]]
+   [text/text {:weight :semi-bold
+               :size   :paragraph-2}
+    (:name (first mentions))]
+   [text/text {:size  :paragraph-2
+               :style {:color        (get-color :text)
+                       :margin-left  3
+                       :margin-right 3}}
+    (i18n/label :added)]
+   [rn/view {:margin-right 4}
+    [user-avatar/user-avatar {:status-indicator? false
+                              :online?           false
+                              :size              :xxxs
+                              :profile-picture   (:image (second mentions))
+                              :ring?             false}]]
+   [text/text {:weight :semi-bold
+               :size   :paragraph-2}
+    (:name (second mentions))]
+   [sm-timestamp timestamp-str]])
+
+(defmethod sm-render :pinned
+  [{:keys [pinned-by content timestamp-str]}]
+  [rn/view {:flex-direction :row
+            :flex           1
+            :align-items    :center}
+   [sm-icon :main-icons/pin20]
+   [rn/view {:flex-direction :column
+             :flex           1}
+    [rn/view {:align-items    :baseline
+              :flex-direction :row}
+     [text/text {:size   :paragraph-2
+                 :weight :semi-bold
+                 :style  {:color (get-color :text)}}
+      pinned-by]
+     [rn/view {:margin-left  4
+               :margin-right 2}
```

Comment:
is this extra view needed?

Reply:
It was added to fine-tune the spacing around the text "pinned a message".